client/src/modules/Admin/Components/Stats.tsx

```diff
@@ -1,194 +1,78 @@
-      })
-  }
+const Stats = ({token}: StatsProps) => {
+    const [approvedRevCount, setApprovedRevCount] = useState<Number>(0)
+    const [pendingRevCount, setPendingRevCount] = useState<Number>(0)
+    const [reportedRevCount, setReportedRevCount] = useState<Number>(0)
+
+    /*
+      Calls route to update the approved, pending, and reported review counts
+      Fires on every render to check for approvals or removals of reviews
+    */
+    useEffect(() => {
+        axios
+        .post('/api/countReviews', {token: token})
+        .then((response) => {
+          const result = response.data.result
+          if (response.status === 200) {
+            setApprovedRevCount(result.approved)
+            setPendingRevCount(result.pending)
+            setReportedRevCount(result.reported)
+          } else {
+            console.log('Error at countReviews')
+          }
+        })
+      }
+    )
 
-  totalReviews() {
-    axios
-      .post(`/api/totalReviews`, {
-        token: this.props.token,
-      })
-      .then((res) => {
-        const total = res.data.result
-        this.setState({ totalReviews: total })
-      })
-      .catch((err) => {
-        console.log('error retrieving totalReviews ', err)
+    /*
+      Function to download a file containing all reviewed classes in the database and their
+      number of reviews
+    */
+    async function downloadCSVFile() {
+      const element = document.createElement('a')
+      let csv = ""
+      
+      await axios
+        .post('/api/getCourseCSV', {token: token})
+        .then((response) => {
+          const result = response.data.result
+          if (response.status === 200) {
+            csv = result
+          } else {
+            console.log('Error at getCourseCSV')
+          }
+        })
+
+      const file = new Blob([csv], {
+        type: 'text/plain',
       })
-  }
-
-  handleClick = (e: any) => {
-    e.preventDefault()
-    this.getChartData()
-  }
-
-  downloadCSVFile = () => {
-    const element = document.createElement('a')
-    const file = new Blob([this.getReviewsPerClassCSV()], {
-      type: 'text/plain',
-    })
-    element.href = URL.createObjectURL(file)
-    element.download = 'ReviewsPerClass.csv'
-    document.body.appendChild(element) // Required for this to work in FireFox
-    element.click()
-  }
+      element.href = URL.createObjectURL(file)
+      element.download = 'ReviewsPerClass.csv'
+      document.body.appendChild(element) // Required for this to work in FireFox
+      element.click()
+    }
 
-  render() {
     return (
-      <div>
-        <div>
-          <button className="" onClick={this.downloadCSVFile}>
-            Download CSV For ReviewsPerClass
+      <div className={styles.diagnosticbox}>
+        <h2>Diagnostic information</h2>
+        <div className = {styles.stats}>
+          <button className={styles.downloadButton} onClick={downloadCSVFile}>
+            Download ApprovedReviewCount by Class
```

Comment:
I would make this a link (like [this](url)) to make it clearer that a file can be downloaded from this text.